### PR TITLE
guessBaseDN: try to parse domain part from a user given in email syntax

### PR DESCRIPTION
**Pull request by jnweiger**
Suggested fix for #797

### CODE REVIEW

File: lib/Wizard.php [hunk 1]
```diff
@@ -685,6 +685,17 @@ public function guessBaseDN() {
 			}
 		}
 
+		//check whether the agent name is in email form
+		$i = \stripos($this->configuration->ldapAgentName, '@');
+		if ($i !== false) {
+			$base1 = \substr($this->configuration->ldapAgentName, $i+1);
+			$base1 = 'dc=' . \implode(',dc=', \explode('.', $base1));
+			if ($this->testBaseDN($base)) {
+				$this->applyFind('ldap_base', $base);
```
Comment (jvillafanez) [inline lib/Wizard.php:694]: I guess this should be `$base1` instead of `$base`

File: lib/Wizard.php [hunk 2]
```diff
@@ -685,6 +685,17 @@ public function guessBaseDN() {
 			}
 		}
 
+		//check whether the agent name is in email form
+		$i = \stripos($this->configuration->ldapAgentName, '@');
```
Comment (jvillafanez) [inline lib/Wizard.php:689]: `strrpos` should work better. We probably should get the position of the last "@" char in case there are multiple

Comment (jnweiger) [reply lib/Wizard.php]: good idea. checking the last never hurts!

File: lib/Wizard.php (hunk 1 above)
Comment (jnweiger) [reply lib/Wizard.php]: oops, you are right, of course.
Thanks. pushed a fix.

File: lib/Wizard.php [hunk 3]
```diff
@@ -685,6 +685,17 @@ public function guessBaseDN() {
 			}
 		}
 
+		//check whether the agent name is in email form; last '@' is taken, if muptiple.
```
Comment (jvillafanez) [inline lib/Wizard.php:688]: tiny typo, not `muptiple`

**Review by jvillafanez — APPROVED** [review]
I'm assuming it's tested and working

**Pull request merged**